provider/datagen/src/provider.rs

```diff
@@ -164,19 +182,23 @@ impl DatagenProvider {
     ///
     /// ✨ *Enabled with the `networking` Cargo feature.*
     #[cfg(feature = "networking")]
-    pub fn with_icuexport_for_tag(self, mut tag: &str) -> Self {
+    pub fn with_icuexport_for_tag(mut self, mut tag: &str) -> Self {
         if tag == "release-71-1" {
             tag = "icu4x/2022-08-17/71.x";
         }
-        Self {
-            source: SourceData {
-                icuexport_paths: Some(Arc::new(SerdeCache::new(AbstractFs::new_from_url(format!(
-                    "https://github.com/unicode-org/icu/releases/download/{tag}/icuexportdata_{}.zip",
-                    tag.replace('/', "-")
-                ))))),
-                ..self.source
-            }
-        }
+        self.source.icuexport_paths = Some(Arc::new(SerdeCache::new(AbstractFs::new_from_url(
+            format!(
+                "https://github.com/unicode-org/icu/releases/download/{tag}/icuexportdata_{}.zip",
+                tag.replace('/', "-")
+            ),
+        ))));
+        self.source.source_versions.icuexport = Some(SourceVersion::Public(
+            tag.strip_prefix("release-")
+                .unwrap_or(tag)
+                .replace('-', ".")
+                .leak(),
```

Comment:
Suggestion: often we use pre-release tags. Maybe just use the whole tag name?

Reply:
The GH tags don't match the version in the ICU repo, so I have to process them somehow. `release-75-rc` will become `75.rc` which I guess is fine. I do want `release-75-1` to become `75.1`, because that's the version that we would read from the files.

Reply:
Why not just read it from the files?

I'm thinking about the tags like `icu4x/2023-12-01/73.1` which I'd like to not need to parse. I'm okayish with bubbling them up to users, though I agree having an actual version number seems more useful.

Reply:
> Why not just read it from the files?

Because then 10 different tags will be `74.1`.

Reply:
How about we save the tag string in `SourceVersions` and then if we want we can add an instance method to `SourceVersions` that parses the ICU version out of the tag.

Reply:
How about you let me parse `release-xx-yy` to `xx.yy`?

Reply:
It seems like a more consistent mental model if `SourceVersions::Public` always contains the tag string.

If you want to parse the release version into `SourceVersions::Public`, you could make a new variant `SourceVersions::Tagged` for arbitrary tags that aren't release tags (including hashes, etc)?